File: algoexpert/python/medium/reverseWordsInString.py
def reverseWordsInString(str):
    if (str == ' '):
        return ' '

    res = ''
    word = ''

    for char in str:
        if (char != ' '):
            word += char
        else:
            res = word + ' ' + res
            word = ''

    res = word + ' ' + res
    return res.rstrip()


print(reverseWordsInString("AlgoExpert is the best!"))
print(reverseWordsInString("test        "))

File: algoexpert/javascript/medium/reverseWordsInString.js
function reverseWordsInString(str) {
  if (str === ' ') return ' ';
  let res = '';
  let word = '';

  for (let char of str) {
    if (char !== " ") {
      word += char;
    } else if (char === " ") {
      res = word + ' ' + res;
      word = '';
    }
  }
  res = word + ' ' + res;
  return res.trimEnd();
}

console.log(reverseWordsInString("AlgoExpert is the best!"));
console.log(reverseWordsInString("test        "));